streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/RackAwareTaskAssignor.java

```diff
@@ -188,7 +188,7 @@ private boolean validateTopicPartitionRack(final boolean changelogTopics) {
      *
      * @return whether the operation successfully completed and the rack information is valid.
      */
-    public static boolean validateTopicPartitionRack(final Cluster fullMetadata,
+    public boolean validateTopicPartitionRack(final Cluster fullMetadata,
```

Comment:
If this method is not static method, we don't need to keep two varieties. Could you please merge two `validateTopicPartitionRack` into single one. Also, the new method can be package-private